src/components/PostsList.tsx

```diff
@@ -16,21 +1,77 @@
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import cl from 'classnames';
 
-      <tbody>
-        <tr data-cy="Post">
+interface Props {
+  posts: Post[];
+  onActivePost: (v: Post | null) => void;
+  onShowSideBar: (v: boolean) => void;
+}
+
+export const PostsList: React.FC<Props> = ({
+  posts,
+  onActivePost,
+  onShowSideBar,
+}) => {
+  const [activePost, setActivePost] = useState<Post | null>(null);
+  const [showClose, setShowClose] = useState(false);
+
+  const handleChangeButton = (post: Post) => {
+    if (post.id === activePost?.id) {
+      setShowClose(!showClose);
+      onShowSideBar(showClose ? false : true);
+
+      return;
+    }
+
+    onActivePost(post);
+    onShowSideBar(true);
+
+    setShowClose(true);
+    setActivePost(post);
+  };
+
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
+
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <th> </th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {posts.map(post => (
+            <tr key={post.id} data-cy="Post">
+              <td data-cy="PostId">{post.id}</td>
+
+              <td data-cy="PostTitle">{post.title}</td>
+
+              <td className="has-text-right is-vcentered">
+                <button
+                  type="button"
+                  data-cy="PostButton"
+                  className={cl('button is-link', {
+                    'is-light': activePost?.id !== post.id || !showClose,
+                  })}
+                  onClick={() => handleChangeButton(post)}
+                >
+                  {showClose && activePost?.id === post.id ? 'Close' : 'Open'}
```

Comment:
Ensure that the button label accurately reflects the current state of the sidebar. The logic here might be confusing if `showClose` is not correctly synchronized with the sidebar state.